node/internal/crypto/cipher.ts

```diff
@@ -0,0 +1,285 @@
+// Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
+// Copyright Joyent, Inc. and Node.js contributors. All rights reserved. MIT license.
+
+import { ERR_INVALID_ARG_TYPE } from "../errors.ts";
+import { validateInt32, validateObject } from "../validators.mjs";
+import { Buffer } from "../../buffer.ts";
+import { notImplemented } from "../../_utils.ts";
+import { TransformOptions } from "../../_stream.d.ts";
+import { KeyObject } from "./keys.ts";
+import type { BufferEncoding } from "../../_global.d.ts";
+import type { BinaryLike, Encoding } from "./types.ts";
+import {
+  privateDecrypt,
+  privateEncrypt,
+  publicDecrypt,
+  publicEncrypt,
+} from "../../_crypto/crypto_browserify/public_encrypt/mod.js";
+
+export {
+  privateDecrypt,
+  privateEncrypt,
+  publicDecrypt,
+  publicEncrypt,
+} from "../../_crypto/crypto_browserify/public_encrypt/mod.js";
+
+export type CipherCCMTypes =
+  | "aes-128-ccm"
+  | "aes-192-ccm"
+  | "aes-256-ccm"
+  | "chacha20-poly1305";
+export type CipherGCMTypes = "aes-128-gcm" | "aes-192-gcm" | "aes-256-gcm";
+export type CipherOCBTypes = "aes-128-ocb" | "aes-192-ocb" | "aes-256-ocb";
+
+export type CipherKey = BinaryLike | KeyObject;
+
+export interface CipherCCMOptions extends TransformOptions {
+  authTagLength: number;
+}
+
+export interface CipherGCMOptions extends TransformOptions {
+  authTagLength?: number | undefined;
+}
+
+export interface CipherOCBOptions extends TransformOptions {
+  authTagLength: number;
+}
+
+export interface Cipher extends TransformStream {
```

Comment:
I guess this should be Node.js `Transform` rather than global `TransformStream`. What do you think?

Reply:
You’re spot on! https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/crypto.d.ts#L850

Will touch up later once back at laptop

Reply:
[85e8a6a](https://github.com/denoland/deno_std/pull/2263/commits/85e8a6a2a2dfe22a777bb138092471ad5ec33b1a)